src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -106,6 +123,10 @@ public bool HasFormContentType
         {
             _parsedFormTask = null;
             _form = value;
+			if (_form is not null && _formContentType is null)
```

Comment:
If we're sticking with a mutable field (see my suggestion above), it seems like we probably want to null out `_formContentType` if `_form` is null?

Reply:
I agree